File: Android/app/src/androidTest/java/com/etiennebaudoux/clipboardzanager/componentmodel/services/ServiceLocatorTests.java
package com.etiennebaudoux.clipboardzanager.componentmodel.services;

import com.etiennebaudoux.clipboardzanager.mocks.ServiceMock;
import com.etiennebaudoux.clipboardzanager.mocks.ServiceSettingProviderMock;

import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertSame;
import static org.junit.Assert.assertTrue;

public class ServiceLocatorTests {
    @Test
    public void serviceLocator() throws Exception {
        ServiceLocator.setSettingProvider(new ServiceSettingProviderMock());

        ServiceMock service1 = ServiceLocator.getService(ServiceMock.class);
        ServiceMock service2 = ServiceLocator.getService(ServiceMock.class);

        assertSame(service1, service2);
        assertFalse(service1.Reseted);

        ServiceLocator.resetAll();
        assertTrue(service1.Reseted);
    }
}

File: ClipboardZanager/Tests/ClipboardZanager.Tests/Services/ServiceLocatorTests.cs
using ClipboardZanager.Shared.Exceptions;
using ClipboardZanager.Tests.Mocks;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace ClipboardZanager.Tests.Services
{
    [TestClass]
    public class ServiceLocatorTests
    {
        [TestMethod]
        public void ServiceLocator()
        {
            Shared.Services.ServiceLocator.SettingProvider = new ServiceSettingProviderMock();
            var service1 = Shared.Services.ServiceLocator.GetService<ServiceMock>();
            var service2 = Shared.Services.ServiceLocator.GetService<ServiceMock>();

            Assert.AreSame(service1, service2);
            Assert.IsFalse(service1.Reseted);

            Shared.Services.ServiceLocator.ResetAll();

            Assert.IsTrue(service1.Reseted);
        }

        [TestMethod]
        [ExpectedException(typeof(NotNullRequiredException))]
        public void ServiceLocatorNoSettingProvider()
        {
            Shared.Services.ServiceLocator.SettingProvider = null;
            var service1 = Shared.Services.ServiceLocator.GetService<ServiceMock>();
        }
    }
}
